src/datapilot/core/platforms/dbt/insights/checks/check_column_desc_are_same.py

```diff
@@ -0,0 +1,106 @@
+from datapilot.core.platforms.dbt.insights.schema import DBTInsightResult
+from datapilot.core.platforms.dbt.insights.schema import DBTModelInsightResponse
+from datapilot.core.platforms.dbt.schemas.manifest import AltimateResourceType
+
+
+class CheckColumnDescAreSame(ChecksInsight):
+    NAME = "Check Column Desc Are Same"
+    ALIAS = "check_column_desc_are_same"
+    DESCRIPTION = (
+        "Checks that models have the same descriptions for the same column names. "
+        "Consistent column descriptions improve understanding and usage of the dbt project."
+    )
+    REASON_TO_FLAG = (
+        "Different descriptions for the same column names can lead to confusion and hinder effective data "
+        "modeling and analysis. It's important to have consistent column descriptions."
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.columns_with_different_desc = []
+        self.columns = {}
+        self.column_node_map = {}
+
+    def _build_failure_result(
+        self,
+    ) -> DBTInsightResult:
+        """
+        Build failure result for the insight if a column has a different description in multiple models or sources.
+
+        :return: An instance of InsightResult containing failure message and recommendation.
+        """
+
+        failure_message = "The following models or sources have different descriptions for some columns:\n"
+        for col_name in self.columns_with_different_desc:
+            failure_message += f"- {self.column_node_map[col_name]} (column: {col_name})\n"
+
+        recommendation = "Ensure that the description for the columns is consistent across all instances."
+
+        return DBTInsightResult(
+            type=self.TYPE,
+            name=self.NAME,
+            message=failure_message,
+            recommendation=recommendation,
+            reason_to_flag=self.REASON_TO_FLAG,
+            metadata={
+                "columns_with_diff_desc": self.columns_with_different_desc,
+            },
+        )
+
+    def generate(self, *args, **kwargs) -> List[DBTModelInsightResponse]:
+        """
+        Generate a list of InsightResponse objects for each model in the DBT project,
+        identifying models with columns that have different descriptions for the same column name.
+        :return: A list of InsightResponse objects.
+        """
+
+        insights = []
+        for node_id, node in self.nodes.items():
+            if self.should_skip_model(node_id):
+                self.logger.debug(f"Skipping model {node_id} as it is not enabled for selected models")
+                continue
+            if node.resource_type == AltimateResourceType.model:
+                self._get_columns_with_different_desc(node_id)
+            elif node.resource_type == AltimateResourceType.source:
+                self._get_columns_with_different_desc(node_id)
+
+        if self.columns_with_different_desc:
```

Comment:
Where are you generasating it? I think it won't ever be created write?

Reply:
And this will prevent any models getting skipped. This should be in the for loop?